pkg/controllers/edgeconnect/controller.go

```diff
@@ -75,52 +83,87 @@ func (controller *Controller) Reconcile(ctx context.Context, request reconcile.R
 		return reconcile.Result{}, nil
 	}
 
-	log.Info("updating version info", "name", request.Name, "namespace", request.Namespace)
+	if edgeConnect.GetDeletionTimestamp() != nil {
+		return reconcile.Result{}, controller.reconcileEdgeConnectDeletion(ctx, edgeConnect)
+	}
 
-	transport := http.DefaultTransport.(*http.Transport).Clone()
-	keyChainSecret := edgeConnect.PullSecretWithoutData()
-	registryClient, err := controller.registryClientBuilder(
-		registry.WithContext(ctx),
-		registry.WithApiReader(controller.apiReader),
-		registry.WithTransport(transport),
-		registry.WithKeyChainSecret(&keyChainSecret),
-	)
+	return controller.reconcileEdgeConnect(ctx, edgeConnect)
+}
+
+func (controller *Controller) reconcileEdgeConnectDeletion(ctx context.Context, edgeConnect *edgeconnectv1alpha1.EdgeConnect) error {
+	log.Info("reconciling EdgeConnect deletion", "name", edgeConnect.Name, "namespace", edgeConnect.Namespace)
+
+	edgeConnect.ObjectMeta.Finalizers = nil
+	if err := controller.client.Update(ctx, edgeConnect); err != nil {
+		return errors.WithStack(err)
+	}
+
+	edgeConnectClient, err := controller.buildEdgeConnectClient(ctx, edgeConnect)
 	if err != nil {
-		return reconcile.Result{}, err
+		return err
 	}
 
-	versionReconciler := version.NewReconciler(controller.apiReader, registryClient, timeprovider.New(), edgeConnect)
-	if err = versionReconciler.Reconcile(ctx); err != nil {
-		log.Error(err, "reconciliation of EdgeConnect failed", "name", request.Name, "namespace", request.Namespace)
-		return reconcile.Result{RequeueAfter: errorUpdateInterval}, nil
+	edgeConnectId, err := controller.getEdgeConnectIdFromClientSecret(ctx, edgeConnect)
+	if err != nil {
+		return err
 	}
 
+	if edgeConnectId == "" {
+		// TODO: managedByDynatraceOperator attribute needed
```

Comment:
will we need `managedByDynatraceOperator` during deletion ?

Reply:
but we are not making a request here, so we would add a request, to delete it by name if it is managed by the operator, right